### PR TITLE
Changing duplicate id behavior for Genbank files. 

### PR DESCRIPTION
**Pull request by slebras**
Dupliate ids are now accepted. the string '.n' is appended where n = "the number of times the id has been previously seen."

### CODE REVIEW

**Comment by jayrbolton** [comment]
Need to update changelog and version

File: lib/GenomeFileUtil/core/GenbankToGenome.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def __init__(self, config):
         self.ont_mappings = load_ontology_mappings('/kb/module/data')
         self.code_table = 11
         self.re_api_url = config.re_api_url
+        self.used_twice_identifiers = {}
```
Comment (jayrbolton) [inline lib/GenomeFileUtil/core/GenbankToGenome.py:66]: Can you add a quick comment here stating the purpose of this var

File: lib/GenomeFileUtil/core/GenbankToGenome.py [hunk 2]
```diff
@@ -726,6 +727,11 @@ def _find_parent_gene(self, potential_id, feature):
             self.defects['bad_parent_loc'] += 1
         return None
 
+    def assign_new_id(self, _id):
```
Comment (jayrbolton) [inline lib/GenomeFileUtil/core/GenbankToGenome.py:730]: Quick docstring? What ID? Gene ID? Contig ID?

File: test/helpdesk_tests/genbank_up_down_duplicate_test.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import os
```
Comment (jayrbolton) [inline test/helpdesk_tests/genbank_up_down_duplicate_test.py:1]: Maybe give a link to the JIRA thread in this module to provide better context

File: test/helpdesk_tests/genbank_up_down_duplicate_test.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        config_file = os.environ.get('KB_DEPLOYMENT_CONFIG', None)
+        cls.cfg = {}
+        config = ConfigParser()
+        config.read(config_file)
+        for nameval in config.items('GenomeFileUtil'):
+            cls.cfg[nameval[0]] = nameval[1]
+        cls.wsURL = cls.cfg['workspace-url']
+        cls.wsClient = workspaceService(cls.wsURL, token=token)
+        cls.serviceImpl = GenomeFileUtil(cls.cfg)
+        gbk_path = "data/Cyanidioschyzon/Cyanidioschyzon_merolae.ASM9120v1.30.dat"
```
Comment (jayrbolton) [inline test/helpdesk_tests/genbank_up_down_duplicate_test.py:35]: Is this actually a genbank file?

File: test/helpdesk_tests/genbank_up_down_duplicate_test.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
+import os
+import time
```
Comment (jayrbolton) [inline test/helpdesk_tests/genbank_up_down_duplicate_test.py:2]: Does this module just test parsing duplicate feature ids? Can it be pared down to just be about that?

File: test/helpdesk_tests/genbank_up_down_duplicate_test.py (hunk 4 above)
Comment (slebras) [reply test/helpdesk_tests/genbank_up_down_duplicate_test.py:35]: yeah, its from Ensembl. I'm not sure why they had it listed as a '.dat' file

**Comment by slebras** [comment]
I set up the new test mainly to have a way to test the current file quickly. Another test module is probably not necessary for this update

I'm not sure how to get around testing this new behavior, other than by running the whole upload. We can definitely tone down the size of the input genbank file for speed

File: lib/GenomeFileUtil/core/GenbankToGenome.py (hunk 2 above)
Comment (slebras) [reply lib/GenomeFileUtil/core/GenbankToGenome.py:731]: added a docstring

File: lib/GenomeFileUtil/core/GenbankToGenome.py (hunk 1 above)
Comment (slebras) [reply lib/GenomeFileUtil/core/GenbankToGenome.py:67]: added comment

File: test/helpdesk_tests/genbank_up_down_duplicate_test.py (hunk 3 above)
Comment (slebras) [reply test/helpdesk_tests/genbank_up_down_duplicate_test.py:4]: added comment

**Comment by jayrbolton** [comment]
I give my thumbs up if it seems not possible to reduce the size of both the test module and test data. I would also reduce all the test assertions to just what you need to prove for this case so it's not redundant. But if those reductions aren't possible then I give thumbs up as-is.

**Comment by jayrbolton** [comment]
What is the plan for this PR? I'm afraid this is going to idle and die, but seems like an important fix.

**Comment by slebras** [comment]
Revisiting this PR. Spent some time thinking about and trying a few things to reduce the size of the test, but ran up against my knowledge of Genbank files

**Comment by jayrbolton** [comment]
LGTM

**Comment by jayrbolton** [comment]
@jsfillman @jkbaumohl You guys have to approve this PR to unblock the merge button because at some point you were set as code owners.

**Pull request merged**